woocommerce/woocommerce-admin
# Onboarding - Profile Wizard: Update plugin installation step to deal with previously installed plugins

Fixes #2599.

This PR updates the main start page and the plugins install page to handle cases where Jetpack and/or WooCommerce Services are already installed or connected. This can happen with a hosting partner, for example.

This PR also updates the activate plugins endpoint. We were running into a race condition activating plugins in multiple requests so quickly. WP's active plugins are stored in a simple serialized array. In some cases, if both plugins were not active. WooCommerce Services would try to activate, and it would have a stale copy of the option, so Jetpack would get wiped out of the active plugins list. My solution for this is to allow activating multiple plugins in one request. Otherwise, we could artificially delay the requests or something else -- I'm open to ideas but think this will work the best.

### Screenshots

<img width="840" alt="Screen Shot 2019-08-21 at 12 20 32 PM" src="https://user-images.githubusercontent.com/689165/63449427-28867f00-c40e-11e9-9dd2-391842f1a19c.png">

### Detailed test instructions:

If Jetpack is in Development mode, it will be considered as connected in these steps. Disable development mode to disconnect.

---

* Deactivate and/or delete Jetpack & WooCommerce Services
* Run through the profile wizard, verify that benefits are shown for both products
* Continue and verify plugins install, activate, and take you to the connection screen

---

* Deactivate and/or delete WooCommerce Services
* Run through the profile wizard, verify that benefits are shown only for WooCommerce Services
* Continue and verify that WooCommerce Services installs and activates. You should be taken to the store details screen

---

* Activate both Jetpack & WooCommerce Services. Make sure Jetpack is connected or in development mode.
* Verify going to the main profiler redirects you to the store details page.
